alpha/apps/kaltura/lib/myPartnerUtils.class.php

```diff
@@ -370,37 +370,131 @@ public static function getCdnHost ( $partner_id, $protocol = null, $hostType = n
 		switch ($hostType)
 		{
 			case 'thumbnail':
-				if ($partner && $partner->getThumbnailHost())
-				{
-					return preg_replace('/^https?/', $protocol, $partner->getThumbnailHost());
-				}
-				if ($partner && $partner->getCdnHost())
-				{
-					return preg_replace('/^https?/', $protocol, $partner->getCdnHost());
-				}
-				return requestUtils::getThumbnailCdnHost($protocol);
+				return self::getThumbnailHostForPartner($partner_id, $protocol);
 			case 'api':
-				if ($protocol == 'https')
-				{
-					$apiHost = (kConf::hasParam('cdn_api_host_https')) ? kConf::get('cdn_api_host_https') : kConf::get('www_host');
-					return 'https://' . $apiHost;
-				}
-				else
+				return self::getApiCdnHost($partner_id, $protocol, kConf::get('www_host'));
+			default:
+				return self::getCdnHostForPartner($partner_id, $protocol);
+		}
+	}
+
+	/**
+	 * @param null $partnerId
+	 * @param string $protocol
+	 * @return string|string[]|null
+	 */
+	public static function getCdnHostForPartner($partnerId = null, $protocol = 'http')
+	{
+		if ($partnerId)
+		{
+			$partner = PartnerPeer::retrieveByPK($partnerId);
+			if($partner && $partner->getCdnHost())
+			{
+				return preg_replace('/^https?/', $protocol, $partner->getCdnHost());
+			}
+		}
+		return requestUtils::getCdnHost($protocol);
+	}
+
+	/**
+	 * @param null $partnerId
+	 * @param null $defaultValue
+	 * @return bool|mixed|string|null
+	 * @throws Exception
+	 */
+	public static function getApiCdnHostHttpsForPartner($partnerId = null, $defaultValue = null)
+	{
+		$host = kConf::hasParam('cdn_api_host_https') ? kConf::get('cdn_api_host_https') : $defaultValue;
```

Comment:
this is not needed , why not do it in function call kConf::get('cdn_api_host_https', mapname,$defaultValue);